src/cc/libbpf.c

```diff
@@ -173,12 +173,22 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
     }
   }
 
-  if (ret < 0 && !log_buf) {
-    // caller did not specify log_buf but failure should be printed,
-    // so call recursively and print the result to stderr
-    bpf_prog_load(prog_type, insns, prog_len, license, kern_version,
-        bpf_log_buf, LOG_BUF_SIZE);
-    fprintf(stderr, "bpf: %s\n%s\n", strerror(errno), bpf_log_buf);
+  if (ret < 0) {
+    const char *str_error = strerror(errno);
```

Comment:
strerror is not thread safe. pls use strerror_r.
Also I don't think the malloc/free pain is worth it.
When there is output in log_buf, it's self explanatory.
Additional errno string will only be confusing,
so I think it makes sense to fill it into the buffer only when log_buf stayed empty after syscall.